Python/wrfrun/unCMIP6.py

```diff
@@ -493,52 +476,61 @@ def interp_data(self):
-                    Temp = Temp/100.0
-                    Temp = Temp.tolist()
-                    Temp = [[round(ii) for ii in nested] for nested in Temp]
-                    Temp = np.array(Temp)                                       
-                    ds.values = Temp
+                    da = (da/100).round()
                 # NOTE: This is because in some model data sftlf is in percentage and WRF expects 0-1 values. 
                 #   Also we do not need to worry about missing values as I checked and there are none of those 
                 #   in the sftlf data.
                 # Handle siconc variable, if needed  
                 if (varname=='siconc' and itm['units']=='%'):
-                    Temp = ds.values
-                    Temp = Temp/100.0
-                    ds.values = Temp
+                    da = da/100.0
                 # NOTE: This is because in some model data siconc is in percentage and WRF expects 0-1 values.
                 #   Also we do not need to worry about missing values as xarray reads those as nans and they  
                 #   remain nans after the above operations.    
                 # Perform the interpolation
                 if lvltype=='3d':
-                    Temp = ds.values
+                    Temp = da.values
                     largetemp = Temp[Temp>1.e10]
-                    n_nanorlarge = np.isnan(Temp).sum()+largetemp.sum()
+                    n_nan = np.isnan(Temp).sum()
+                    n_large = largetemp.sum()
+                    n_nanorlarge = n_nan+n_large
                     if n_nanorlarge==0:                        
-                        self.outfrm[varname] = ds.copy(deep=True)                            
-                    else:
-                        raise ValueError("Error: There are nans or extremely large values in the 3D data.")                    
+                        self.outfrm[varname] = da.copy(deep=True)                            
+                    else:                        
+                        if n_large>0:
+                            raise ValueError("Error: There are extremely large values in the 3D data.")
+                        else:
+                            K = da.shape[0]
+                            da_f = da.copy(deep=True)
+                            for k in range(K):
+                                da_l = da[k,:,:].copy(deep=True)
+                                n_nan_l = np.isnan(da_l).sum()
+                                if n_nan_l>0:
+                                    array = np.ma.masked_invalid(da_l.values)
+                                    xx,yy = np.meshgrid(self.lon,self.lat)
+                                    xx1 = xx[~array.mask]
+                                    yy1 = yy[~array.mask]
+                                    array1 = array[~array.mask]
+                                    da_f[k,:,:] = griddata((xx1,yy1),array1,(xx,yy),method='nearest')
```

Comment:
I guess this is to interpolate NaNs in pressure levels? `griddata` is extremely inefficient for this purpose, since afaik it does not assume any structure (like a point cloud of station values), but we have a lot of structure here.
I think a 1D interpolation along a horizontal axis would probably suffice (as those values should not really be used by metgrid anyway).
That being said, is this in any way a bottleneck? (i.e. how much time does this take, compared to the overall runtime)

Reply:
I checked and thsi is not a bottleneck. So I will leave it as is, as we have been using this all along and changing it might cause inconsistencies.